packages/mui-base/src/unstable_useModal/useModal.ts

```diff
@@ -0,0 +1,239 @@
+'use client';
+import * as React from 'react';
+import {
+  unstable_ownerDocument as ownerDocument,
+  unstable_useForkRef as useForkRef,
+  unstable_useEventCallback as useEventCallback,
+  unstable_createChainedFunction as createChainedFunction,
+} from '@mui/utils';
+import { EventHandlers, extractEventHandlers } from '../utils';
+import { ModalManager, ariaHidden } from './ModalManager';
+import {
+  UseModalParameters,
+  UseModalReturnValue,
+  UseModalRootSlotProps,
+  UseModalBackdropSlotProps,
+} from './useModal.types';
+
+function getContainer(container: UseModalParameters['container']) {
+  return typeof container === 'function' ? container() : container;
+}
+
+function getHasTransition(children: UseModalParameters['children']) {
+  return children ? children.props.hasOwnProperty('in') : false;
+}
+
+// A modal manager used to track and manage the state of open Modals.
+// Modals don't open on the server so this won't conflict with concurrent requests.
+const defaultManager = new ModalManager();
+/**
+ *
+ * Demos:
+ *
+ * - [Modal](https://mui.com/base-ui/react-modal/#hook)
+ *
+ * API:
+ *
+ * - [useModal API](https://mui.com/base-ui/react-modal/hooks-api/#use-modal)
+ */
+export function useModal(parameters: UseModalParameters): UseModalReturnValue {
+  const {
+    container,
+    disableEscapeKeyDown = false,
+    disableScrollLock = false,
+    // @ts-ignore internal logic - Base UI supports the manager as a prop too
+    manager = defaultManager,
+    closeAfterTransition = false,
+    onTransitionEnter,
+    onTransitionExited,
+    children,
+    onClose,
+    open,
+    rootRef,
+  } = parameters;
+
+  // @ts-ignore internal logic
+  const modal = React.useRef<{ modalRef: HTMLDivElement; mount: HTMLElement }>({});
+  const mountNodeRef = React.useRef<null | HTMLElement>(null);
+  const modalRef = React.useRef<null | HTMLDivElement>(null);
+  const handleRef = useForkRef(modalRef, rootRef);
+  const [exited, setExited] = React.useState(!open);
+  const hasTransition = getHasTransition(children);
+
+  let ariaHiddenProp = true;
+  if (
+    parameters['aria-hidden'] === 'false' ||
+    (typeof parameters['aria-hidden'] === 'boolean' && !parameters['aria-hidden'])
```

Comment:
Wouldn't `parameters['aria-hidden'] === false` suffice?

Reply:
Yes, I will change it. The logic was like this before, haven't noticed it 👍